### PR TITLE
Add support for slacklining

### PR DESCRIPTION
**Pull request by tsauerwein**
See https://github.com/c2corg/v6_ui/issues/1543
Requires https://github.com/c2corg/v6_api/pull/662 and https://github.com/c2corg/v6_common/pull/56

### CODE REVIEW

**Comment by asaunier** [comment]
Hi @tsauerwein 
Does it means the PR is now ready for review?

File: c2corg_ui/static/partials/cards/routes.html [hunk 1]
```diff
@@ -23,7 +23,17 @@
             <span class="glyphicon glyphicon-resize-vertical" ng-if="::doc['height_diff_difficulties']"></span>
             <span class="value" ng-if="::doc['height_diff_difficulties']">{{::doc['height_diff_difficulties']}}&nbsp;m</span>
           </span>
-          <div  class="orientations"
+          <span class="slackline-length" ng-if="::doc['route_length'] && cardCtrl.hasActivity(['slacklining'])"
```
Comment (asaunier) [inline c2corg_ui/static/partials/cards/routes.html]: @stef74 Wouldn't this ``cardCtrl.hasActivity()`` method be useful in your work about card contents?

File: c2corg_ui/static/js/lengthconverter.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+      ngModel.$formatters.push(function(value) {
+        return value / 1000;
+      });
+    }
+  };
+};
+
+
+app.module.directive('lengthConverter', app.lengthConverterDirective);
```
Comment (asaunier) [inline c2corg_ui/static/js/lengthconverter.js]: What about using the ``app`` prefix as for most (all?) other project-specific directives in https://github.com/c2corg/v6_ui/tree/master/c2corg_ui/static/js
It helps figuring out where this directive comes from.

File: c2corg_ui/templates/route/helpers/view.html [hunk 3]
```diff
@@ -170,7 +173,11 @@
        ${get_document_up_down(route, 'height_diff')}
 
         % if route.get('route_length'):
-        <p><span translate class="value-title">route_length</span>: <b class="value">${route['route_length']/1000}&nbsp;km</b></p>
+        <p><span translate class="value-title">route_length</span>: <b class="value">${format_length(route['route_length']) | n}</b></p>
```
Comment (asaunier) [inline c2corg_ui/templates/route/helpers/view.html]: Should you use ``format_length`` for attributes ``route['mtb_length_asphalt']`` and ``route['mtb_length_trail']`` a few lines below?

File: c2corg_ui/templates/route/filters.html [hunk 4]
```diff
@@ -37,6 +38,18 @@
...
 
+      ## Slacklining
+      <div ng-show="filtersCtrl.filters.act.length && filtersCtrl.filters.act.indexOf('slacklining') !== -1">
+        <div class="col-xs-12 col-sm-6 filter">
+          <label translate>slackline_type</label><br>
+          ${add_multiselect('sltyp', slackline_types)}
+        </div>
+        <div class="col-xs-12 col-sm-6 filter">
+          <label translate>Length</label><br>
+          <app-slider filter="rlen" filters-list="filtersCtrl.filters" max="2000" step="10" unit="m"></app-slider>
```
Comment (asaunier) [inline c2corg_ui/templates/route/filters.html]: 2000m? Wow :)
Are there really 2000m-long lines or are you just anticipating the records coming soon?

**Comment by asaunier** [comment]
A few comments while testing the branch in my instance:
* the slacklining icon is nice :)
* when editing a "slacklining" route, you have a "slacklining" > "length" attribute as well as "caracteristics" > "route length", both automatically sharing the same value (the first one in meters, the second one in kilometers). Is it on purpose? Isn't it a duplicate?
* it is possible to precise what kind of routes are available in a given "slacklining_spot" WP (slackline, highline, waterline). A bit like the "outdoor_climbing_site_type" (don't remember the name) for climbing site WP. But it seems it is not possible to filter on such types when selecting the "slacklining_spot" WP type. A possible improvement for the future I guess :)

It seems there's also a conflict in the stylesheets.

"Else" that looks great to me!

File: c2corg_ui/templates/route/filters.html (hunk 4 above)
Comment (tsauerwein) [inline c2corg_ui/templates/route/filters.html]: The current world record is [1020m](https://www.youtube.com/watch?v=2924rs9e4QE), and there are already plans to double the distance this year. Crazy! :)

**Comment by tsauerwein** [comment]
> when editing a "slacklining" route, you have a "slacklining" > "length" attribute as well as "caracteristics" > "route length", both automatically sharing the same value (the first one in meters, the second one in kilometers). Is it on purpose? Isn't it a duplicate?

I wanted to have the fields related to the slackline topo together at one place. If you want I can hide the 2nd field under "Caractéristiques" when "slacklining" is selected.

**Comment by tsauerwein** [comment]
> it is possible to precise what kind of routes are available in a given "slacklining_spot" WP (slackline, highline, waterline). A bit like the "outdoor_climbing_site_type" (don't remember the name) for climbing site WP. But it seems it is not possible to filter on such types when selecting the "slacklining_spot" WP type. A possible improvement for the future I guess :)

Yes, that could be an improvement. I think most people would directly search for routes.

> It seems there's also a conflict in the stylesheets.

Resolved.

Thanks for the reviews!

**Comment by asaunier** [comment]
> I wanted to have the fields related to the slackline topo together at one place. If you want I can hide the 2nd field under "Caractéristiques" when "slacklining" is selected.

I don't have a strong opinion about that. I just find a bit surprising to have 2 fields for the same attribute (the simpler the form the better I think). But if you think it makes sense then we can leave it as is.

File: c2corg_ui/templates/route/helpers/view.html (hunk 3 above)
Comment (asaunier) [inline c2corg_ui/templates/route/helpers/view.html]: Do you plan to make these changes as well?

Comment (asaunier) [inline c2corg_ui/templates/route/helpers/view.html]: Oops, you did. Sorry :P

**Pull request merged**